src/Foundation/Workflow/code/Actions/SubmissionNotification.cs

```diff
@@ -10,124 +10,129 @@
-                smtpClient.Send(mailMessage);
-            }
-            catch (Exception ex)
-            {                      
-                Log.Error($"Failed to send mail: {ex.Message}", ex, typeof(SubmissionNotification));
-            }
-        }
-
-        /// <summary>
-        /// Gets the text.
-        /// </summary>
-        /// <param name="commandItem">The command item.</param>
-        /// <param name="field">The field.</param>
-        /// <param name="args">The arguments.</param>
-        /// <returns></returns>
-        private string GetText(Item commandItem, string field, WorkflowPipelineArgs args)
-        {
-            string text = commandItem[field];
-            if (text.Length > 0)
-            {
-                return this.ReplaceVariables(text, args);
-            }
-            return string.Empty;
-        }
-
-        /// <summary>
-        /// Replaces the variables.
-        /// </summary>
-        /// <param name="text">The text.</param>
-        /// <param name="args">The arguments.</param>
-        /// <returns></returns>
-        private string ReplaceVariables(string text, WorkflowPipelineArgs args)
-        {
-            Item contentItem = args.DataItem;
-
-            var contentWorkflow = contentItem.Database.WorkflowProvider.GetWorkflow(contentItem);
-            var contentHistory = contentWorkflow.GetHistory(contentItem);
-
-            if (contentHistory.Length > 0)
-            {
-                var lastUser = contentHistory.Last().User;
-                text = text.Replace("$submitter$", lastUser);
-            }
-
-            text = text.Replace("$itemPath$", args.DataItem.Paths.FullPath);
-            text = text.Replace("$itemLanguage$", args.DataItem.Language.ToString());
-            text = text.Replace("$itemVersion$", args.DataItem.Version.ToString());
-            text = text.Replace("$hostname$", HttpContext.Current.Request.Url.Host);
-            text = text.Replace ( "$comment$" , args.CommentFields[ "Comments" ] );
-
-
-            return text;
-        }
-
-        public static MailMessage AddRecipientsToMail(MailMessage mailMessage, string roleName)
-        {      
-            if (Role.Exists(roleName))
-            {
-                Role role = Role.FromName(roleName);
-                List<User> users = RolesInRolesManager.GetUsersInRole(role, true)
-                    .Where(x => x.IsInRole(role))
-                    .Where(user => !string.IsNullOrEmpty(user.Profile.Email))
-                    .ToList();
-
-                foreach (User user in users)
-                {
-                    mailMessage.To.Add(user.Profile.Email);
-                }
-            }
-            else
-            {
-                Log.Error($"No Users with valid email addresses found in role {roleName}", typeof(SubmissionNotification));
-            }
-
-            return mailMessage;
-        }
-    }
-}
\ No newline at end of file
+	public class SubmissionNotification
```

Comment:
All 4 spaces indentation were replaced by tab characters in this file. Could you please revert to 4 spaces indentation?

Reply:
This is still an issue with the pull request. When you used the Visual Studio extension to remove trailing whitespace at the end of lines, the file was formatted with whatever formatting rules are configured in your copy of Visual Studio. This changed the indentation of all the lines of this file from "4 spaces" to "tab" characters. Please convert the lines indentation back to "4 spaces" instead of "tab" characters. Thanks

Reply:
will do .